src/Functions/DateTimeTransforms.h

```diff
@@ -1425,19 +1428,49 @@ struct ToDateTimeComponentsImpl
     using FactorTransform = ZeroTransform;
 };
 
+struct DateTimeAccurateConvertStrategyAdditions {};
+struct DateTimeAccurateOrNullConvertStrategyAdditions {};
 
-template <typename FromType, typename ToType, typename Transform, bool is_extended_result = false>
+template <typename FromType, typename ToType, typename Transform, bool is_extended_result = false, typename Additions = void *>
 struct Transformer
 {
     template <typename FromTypeVector, typename ToTypeVector>
-    static void vector(const FromTypeVector & vec_from, ToTypeVector & vec_to, const DateLUTImpl & time_zone, const Transform & transform)
+    static void vector(const FromTypeVector & vec_from, ToTypeVector & vec_to, const DateLUTImpl & time_zone, const Transform & transform,
+        ColumnUInt8::Container * vec_null_map_to [[maybe_unused]])
     {
         using ValueType = typename ToTypeVector::value_type;
         size_t size = vec_from.size();
         vec_to.resize(size);
 
         for (size_t i = 0; i < size; ++i)
         {
+            if constexpr (std::is_same_v<Additions, DateTimeAccurateConvertStrategyAdditions>
+                || std::is_same_v<Additions, DateTimeAccurateOrNullConvertStrategyAdditions>)
+            {
+                bool check_range_result = true;
+
+                if constexpr (std::is_same_v<ToType, DataTypeDate> || std::is_same_v<ToType, DataTypeDateTime>)
+                {
```

Comment:
(cosmetic: I don't mind too much but could we generally omit parentheses for single line if statements, similar to l. 1474ff?)